src/lib/fpti.js

```diff
@@ -0,0 +1,140 @@
+/* @flow */
+import { getClientID, getMerchantID, getPartnerAttributionID } from '@paypal/sdk-client/src';
+
+import { getDeviceInfo } from './get-device-info';
+
+const sendBeacon = (src, data) => {
+    if (!src || !data) {
+        return;
+    }
+
+    let query = [];
+
+    for (const key in data) {
+        if (data.hasOwnProperty(key)) {
+            query.push(`${ encodeURIComponent(key) }=${ encodeURIComponent(data[key]) }`);
+        }
+    }
+
+    query = query.join('&');
+
+    const beaconImage = new window.Image();
+    beaconImage.src = `${ src }?${ query }`;
+};
+
+// removes empty strings, `undefined`, `null`, and `NaN` from fpti event
+const filterFalsyValues = source => {
+    const result = {};
+
+    for (const key in source) {
```

Comment:
https://stackoverflow.com/questions/30326452/why-is-object-keys-faster-than-hasownproperty

Why not use `Object.keys(source)` instead of a `for` and `hasOwnProperty`? Applies to all functions.